### PR TITLE
Fundamental Matrix and misc. cleanup

### PR DESCRIPTION
**Pull request by jlaura**

### CODE REVIEW

**Comment by jlaura** [comment]
@kree if you do no have time to review, let me know and I can merge the PR

**Comment by jlaura** [comment]
Something whacky went on with the build - I'll comment again when I figure out what is going on.

File: autocnet/vis/graph_view.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import inspect
```
Comment (kree) [inline autocnet/vis/graph_view.py]: Was this added for debugging purposes? Should this stay in here?

File: functional_tests/test_two_image.py [hunk 2]
```diff
@@ -66,11 +66,11 @@ def test_two_image(self):
 
             # Perform the symmetry check
             edge.symmetry_check()
-            self.assertIn(edge.masks[0]['symmetry'].sum(), range(430, 461))
+            self.assertIn(edge.masks['symmetry'].sum(), range(430, 461))
```
Comment (kree) [inline functional_tests/test_two_image.py]: :+1:  I like this a lot better.

File: autocnet/matcher/homography.py [hunk 3]
```diff
@@ -4,126 +4,5 @@
 from autocnet.utils.utils import make_homogeneous
```
Comment (kree) [inline autocnet/matcher/homography.py]: Should homography.py even still exist in this PR? It looks like maybe it should just be deleted/removed now that transformations.py has replaced it?

File: autocnet/graph/tests/test_network.py [hunk 4]
```diff
@@ -46,7 +48,8 @@ def test_connected_subgraphs(self):
     def test_save_load(self):
         self.graph.save('test_save.cg')
         loaded = self.graph.from_graph('test_save.cg')
-
+        for s, d, e in loaded.edges_iter(data=True):
+            print(s, d)
```
Comment (kree) [inline autocnet/graph/tests/test_network.py]: Do you want to leave this print statement in?

File: autocnet/graph/edge.py [hunk 5]
```diff
@@ -362,9 +359,7 @@ def _clean(self, clean_keys, pid=None):
         mask : series
                     A boolean series to inflate back to the full match set
         """
-        if not pid:
-            pid = self._pid
-        panel = self.masks[pid]
+        panel = self.masks
         mask = panel[clean_keys].all(axis=1)
         matches = self.matches[mask]
         return matches, mask
```
Comment (kree) [inline autocnet/graph/edge.py]: PEP wants a newline

File: autocnet/graph/edge.py [hunk 6]
```diff
@@ -49,34 +47,25 @@ def __repr__(self):
 
     @property
     def masks(self):
+        mask_lookup = {'fundamental': 'fundamental_matrix'}
         if not hasattr(self, '_masks'):
-            self._masks = pd.Panel({self._pid: pd.DataFrame(index=self.matches.index)})
+            if hasattr(self, 'matches'):
+                self._masks = pd.DataFrame(True, columns=['symmetry', 'ratio'],
```
Comment (kree) [inline autocnet/graph/edge.py]: Why do you populate this with all True values if matches exists?

File: autocnet/matcher/transformations.py [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+
+import numpy as np
+import pandas as pd
+import pysal as ps
+
+from autocnet.matcher.outlier_detector import compute_fundamental_matrix
+from autocnet.utils.utils import make_homogeneous
+
+class TransformationMatrix(np.ndarray):
+    __metaclass__ = abc.ABCMeta
```
Comment (kree) [inline autocnet/matcher/transformations.py]: :+1: This looked really strange until I realized that abc = "abstract base class"

**Comment by kree** [comment]
Okay. I'm still looking at the changes in transformations.py, but if I take too long and you want to go ahead and merge it, feel free.

File: autocnet/matcher/health.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        """
+        for a in args:
+            if hasattr(self, a.__class__.__name__):
+                setattr(self, a.__class__.__name__, a)
+
+    def recompute_health(self):
+        """
+        Recompute the health of the edge.
+        """
+        return self.FundamentalMatrix.error.mean()
```
Comment (kree) [inline autocnet/matcher/health.py]: PEP wants a newline

**Pull request merged**